### PR TITLE
setup basic bdd test and ui for create

### PR DESCRIPTION
**Pull request by bf2285**

### CODE REVIEW

File: features/steps/steps.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    context.driver.get(context.BASE_URL)
+
+@then('I should see "{message}" in the title')
+def step_impl(context, message):
+    """ Check the document title for a message """
+    expect(context.driver.title).to_contain(message)
+
+@then('I should not see "{message}"')
+def step_impl(context, message):
+    assert message not in str(context.resp.text)
```
Comment (MahamArif) [inline features/steps/steps.py:46]: Can you replace this with the following code from professor's example?

```
@then('I should not see "{text_string}"')
def step_impl(context, text_string):
    element = context.driver.find_element(By.TAG_NAME, 'body')
    error_msg = "I should not see '%s' in '%s'" % (text_string, element.text)
    ensure(text_string in element.text, False, error_msg)
```

File: features/steps/steps.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# These two function simulate copy and paste
+##################################################################
+@when('I copy the "{element_name}" field')
+def step_impl(context, element_name):
+    element_id = ID_PREFIX + element_name.lower().replace(' ', '_')
+    element = WebDriverWait(context.driver, context.WAIT_SECONDS).until(
+        expected_conditions.presence_of_element_located((By.ID, element_id))
+        )
+    context.clipboard = element.get_attribute('value')
+    logging.info('Clipboard contains: %s', context.clipboard)
```
Comment (MahamArif) [inline features/steps/steps.py:99]: Add new line to end of file

File: service/static/index.html [hunk 3]
```diff
@@ -13,6 +13,79 @@
...
+              </div>
+            </div>
+
+            <!-- UPDATED ON -->
+            <div class="form-group">
+              <label class="control-label col-sm-2" for="order_updated_on">Gender:</label>
+              <div class="col-sm-10">
+                <input type="text" class="form-control" id="order_updated_on" readonly>
+              </div>
+            </div>
```
Comment (MahamArif) [inline service/static/index.html:75]: Please remove these fields from the form, these are internal database fields and should not be displayed on UI.

File: service/static/js/rest_api.js [hunk 4]
```diff
@@ -1,3 +1,72 @@
...
+    //  U T I L I T Y   F U N C T I O N S
+    // ****************************************
 
+    // Updates the form with data from the response
+    function update_form_data(res) {
+        $("#order_id").val(res.id);
+        $("#order_customer_id").val(res.customer_id);
+        $("#order_status").val(res.status);
+        $("#order_created_on").val(res.created_on);
+        $("#order_updated_on").val(res.updated_on);
```
Comment (MahamArif) [inline service/static/js/rest_api.js:12]: Please remove these two fields

File: features/steps/steps.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    """ Delete all Orders and load new ones """
+    # List all of the orders and delete them one by one
+    rest_endpoint = f"{context.BASE_URL}/orders"
+    context.resp = requests.get(rest_endpoint)
+    expect(context.resp.status_code).to_equal(200)
+    for order in context.resp.json():
+        context.resp = requests.delete(f"{rest_endpoint}/{order['id']}")
+        expect(context.resp.status_code).to_equal(204)
+
+    # load the database with new pets
```
Comment (MahamArif) [inline features/steps/steps.py:25]: change to comment to say orders instead of pets

File: service/static/js/rest_api.js [hunk 6]
```diff
@@ -1,3 +1,72 @@
...
+    }
+
+    // Updates the flash message area
+    function flash_message(message) {
+        $("#flash_message").empty();
+        $("#flash_message").append(message);
+    }
+
+    // ****************************************
+    // Create a Pet
```
Comment (MahamArif) [inline service/static/js/rest_api.js:30]: Change the comment to Create an Order

**Review by MahamArif — CHANGES_REQUESTED** [review]

File: features/orders.feature [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    And I should not see "404 Not Found"
+
+Scenario: Create an Order
+    When I visit the "Home Page"
+    And I set the "Customer ID" to "5"
+    And I select "Confirmed" in the "Status" dropdown
+    And I press the "Create" button
+    Then I should see the message "Success"
+    And I should see "Today's date" in the "Created On" field
+    And I should see "Today's date" in the "Updated On" field
```
[the remark this answers is not in the document]
Comment (bf2285) [reply features/orders.feature:28]: I have different thoughts on these two fields. Since the UI site is for the order manager, this would be part of the information they want to see. I made these two fields read-only in the UI so it is not modifiable, but still provides the information. Let me know that you think

File: features/orders.feature [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+Scenario: The server is running
+    When I visit the "Home Page"
+    Then I should see "Order Demo RESTful Service" in the title
+    And I should not see "404 Not Found"
+
+Scenario: Create an Order
+    When I visit the "Home Page"
+    And I set the "Customer ID" to "5"
+    And I select "Confirmed" in the "Status" dropdown
```
[the remark this answers is not in the document]
Comment (bf2285) [reply features/orders.feature:24]: It won't matter since deserialization will just enum if there is a status value. There's no difference if it is the default, but sure I can change that. I can also add a test where only Customer ID is provided to test for default.

File: features/steps/steps.py (hunk 1 above)
Comment (bf2285) [reply features/steps/steps.py:46]: Will do.

File: features/steps/steps.py (hunk 2 above)
Comment (bf2285) [reply features/steps/steps.py:99]: Will do

File: features/orders.feature (hunk 7 above)
[the remark this answers is not in the document]
Comment (MahamArif) [reply features/orders.feature:28]: I see, you can leave these fields unchanged. But the title for the updated on field says Gender. Please correct that in HTML.

File: features/orders.feature (hunk 8 above)
[the remark this answers is not in the document]
Comment (MahamArif) [reply features/orders.feature:24]: It is fine to just change the value over here, no need to add a new test. Integration tests are just for testing the overall behavior. The rigorous testing for corner cases is already done by unit tests.

File: features/steps/steps.py (hunk 5 above)
Comment (bf2285) [reply features/steps/steps.py:25]: Will do

File: service/static/js/rest_api.js (hunk 6 above)
Comment (bf2285) [reply service/static/js/rest_api.js:30]: Will do

File: features/orders.feature (hunk 7 above)
[the remark this answers is not in the document]
Comment (bf2285) [reply features/orders.feature:28]: Will do

**Review by MahamArif — APPROVED** [review]
